src/translate/pddl_parser/parsing_functions.py

```diff
@@ -188,121 +314,229 @@ def add_effect(tmp_effect, result):
+                    effect_tag = next(iterator)
+                else:
+                    precondition = pddl.Conjunction([])
+                    effect_tag = precondition_tag_opt
+            with context.layer("Parsing effect"):
+                context.check_string(effect_tag, value=":effect", item_name="Effect tag",
+                                     finalmsg=f"Expecting '{syntax}'.")
+                effect_list = next(iterator)
+                context.check_list(effect_list, block_name="Effect")
+                eff = []
+                if effect_list:
+                    try:
+                        cost = parse_effects(
+                            context, effect_list, eff, type_dict, predicate_dict)
+                    except ValueError as e:
+                        context.error("Error in Action %s\nReason: %s." % (name, e))
+        except StopIteration:
+            context.error(f"Missing fields. Expecting {syntax}.")
+
     if eff:
         return pddl.Action(name, parameters, len(parameters),
                            precondition, eff, cost)
     else:
         return None
 
 
-def parse_axiom(alist, type_dict, predicate_dict):
-    assert len(alist) == 3
-    assert alist[0] == ":derived"
-    predicate = parse_predicate(alist[1])
-    condition = parse_condition(
-        alist[2], type_dict, predicate_dict)
-    return pddl.Axiom(predicate.name, predicate.arguments,
-                      len(predicate.arguments), condition)
+def parse_axiom(context, no, alist, type_dict, predicate_dict):
+    syntax = "'(:derived PREDICATE CONDITION)'"
+    with context.layer(f"Parsing {no}. axiom"):
+        if len(alist) != 3:
+            context.error(f"Expecting {syntax}.")
+        assert alist[0] == ":derived"
+        context.check_list(
+            alist[1], block_name="PREDICATE",
+            finalmsg=f"Expecting {syntax}.")
+        predicate = parse_predicate(context, 1, alist[1])
+        context.check_list(
+            alist[2], block_name="CONDITION",
+            finalmsg=f"Expecting {syntax}.")
+        condition = parse_condition(
+            context, alist[2], type_dict, predicate_dict)
+        return pddl.Axiom(predicate.name, predicate.arguments,
+                          len(predicate.arguments), condition)
+
+
+def parse_init(context, alist):
+    with context.layer("Parsing init"):
+        context.check_expected_block(alist, [":init"])
+    initial = []
+    initial_true = set()
+    initial_false = set()
+    initial_assignments = dict()
+    for no, fact in enumerate(alist[1:], 1):
+        with context.layer(f"Parsing {no}. element in init block"):
+            context.check_list(fact, min_size=1)
+            if fact[0] == "=":
+                try:
+                    assignment = parse_assignment(context, fact)
+                except ValueError as e:
+                    context.error(f"Error in initial state specification\n"
+                                  f"Reason: {e}.")
```

Comment:
There is a mix of format strings an `%` syntax in this file. I'd either change the other places or leave this one.